zebra-chain/src/note_encryption.rs

```diff
@@ -0,0 +1,90 @@
+//! Note encryption types.
+
+use std::{
+    convert::{AsRef, From},
+    fmt,
+};
+
+mod sapling;
+mod sprout;
+
+/// A 512-byte _Memo_ field associated with a note, as described in
+/// [protocol specification §5.5][ps].
+///
+/// The usage of the memo field is by agreement between the sender and
+/// recipient of the note.
```

Comment:
It might be good to mention in the `Memo` docs that an encryption of the memo is committed to the chain, so that it's clear that the `Memo` type is a kind of plaintext.

Reply:
Yah very good point.